test/test_inference/test_signature.py

```diff
@@ -82,6 +95,11 @@ def func(a, b, c):
         (partial_code + 'b(', 'func(b, c)'),
         (partial_code + 'c(', 'func(b)'),
         (partial_code + 'd(', None),
+
+        (partialmethod_code + 'X().a(', 'func(a, b, c)'),
+        (partialmethod_code + 'X().b(', 'func(b, c)'),
+        (partialmethod_code + 'X().c(', 'func(b)'),
+        (partialmethod_code + 'X().d(', None),
```

Comment:
Good tests! Please also add a test for `X.c(` and `X.d`, just to make sure that stuff doesn't raise exceptions. I'm not really interested that it behaves correctly, it's more about avoiding exceptions.

Reply:
I've added the following:
```python
(partialmethod_code + 'X.c(', 'func(b)'),
(partialmethod_code + 'X.d(', None),
```
Although I think `X.c(` should technically include `self` in the signature

Reply:
Doesn't really matter, nobody is going to use it like that anyway :) That might start working if we get the instance stuff working.